beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/historical/HistoricalBlockSyncService.java

```diff
@@ -137,6 +145,9 @@ protected SafeFuture<?> doStop() {
     LOG.debug("Stop {}", getClass().getSimpleName());
     syncStateProvider.unsubscribeFromSyncStateChanges(syncStateSubscription.get());
     badPeerCache.clear();
+    reconstructHistoricalStatesService
+        .doStop()
+        .finish(error -> LOG.error(error.getMessage())); // fixme log error not working
     return SafeFuture.COMPLETE;
```

Comment:
I think in this case we should be returning the result of doStop():
```suggestion
    return reconstructHistoricalStatesService
        .stop();
```

because the `HistoricalBlockSyncService` is entirely encapsulating the `ReconstructHistoricalStatesService` so historical block service isn't stopped until historical states service is.  I suspect though you should be calling `.stop()` not `.doStop()`.

The .finish looks right and should log fine - it will only go to the log file by default, not the console though...

Reply:
Do we mind that the log only goes into the log file, or should it go through the console too?

Reply:
I think in this case you'll just return the future so won't need the finish.  If the `start()` call fails we probably should ensure the log goes to the console though.